resources/lib/utilities.py

```
@@ -215,6 +215,31 @@ def findEpisodeMatchInList(id, seasonNumber, episodeNumber, list, idType):
 
 def convertDateTimeToUTC(toConvert):
     if toConvert:
+        # Check if already a datetime object
+        if isinstance(toConvert, datetime):
+            try:
+                if toConvert.tzinfo is None:
+                    local = toConvert.replace(tzinfo=tzlocal())
+                else:
+                    local = toConvert
+                utc = local.astimezone(tzutc())
+                return str(utc)
+            except (ValueError, AttributeError) as e:
+                logger.debug(
+                    "convertDateTimeToUTC() ValueError/AttributeError with datetime object: %s. Fallback to datetime utcnow"
+                    % str(e)
+                )
+                return str(datetime.utcnow())
+        
+        # Check if it's a timedelta (should not happen, but handle gracefully)
+        if isinstance(toConvert, timedelta):
+            logger.error(
+                "convertDateTimeToUTC() received timedelta object instead of datetime/string: %s. This is a bug. Fallback to datetime utcnow"
+                % str(toConvert)
+            )
+            return str(datetime.utcnow())
```

Not sure I want this to not fail hard. This would cause watched timestamps and added at timestamps to be now instead of the correct date.